core/src/main/scala/org/apache/spark/storage/StorageUtils.scala

```diff
@@ -193,6 +194,36 @@ private[spark] class StorageStatus(
 
 /** Helper methods for storage-related objects. */
 private[spark] object StorageUtils extends Logging {
+
+  // In Java 8, the type of DirectBuffer.cleaner() was sun.misc.Cleaner, and it was possible
+  // to access the method sun.misc.Cleaner.clean() to invoke it. The type changed to
+  // jdk.internal.ref.Cleaner in later JDKs, and the .clean() method is not accessible even with
+  // reflection. However sun.misc.Unsafe added a invokeCleaner() method in JDK 9+ and this is
+  // still accessible with reflection.
+  private val bufferCleaner: DirectBuffer => Unit =
+    // Split java.version on non-digit chars:
+    if (System.getProperty("java.version").split("\\D+").head.toInt < 9) {
```

Comment:
Is this java detection method used elsewhere? Maybe other parts of the codebase could benefit from this.

Reply:
In the Platform.java class we can't rely on any other libs or methods, so I wrote something like this. I just followed suit here. My reasoning was that this is a simple check of major version number, specific to 8 vs 9. I don't mind using a library method here, but we can't in the same context in Platform.java.

Reply:
I think so how about then:
    import org.apache.commons.lang3.JavaVersion
    SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_9)
We include that package anyway, and that method is based on that property:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/JavaVersion.html

Reply:
I like it. I made both changes. Actually there's just one other location where java.version is parsed where we can use lang3, so I just changed that too. That's more convincing as it's more consistent.